mysql/MySqlParser.g4

```diff
@@ -1994,14 +1994,22 @@ dataType
       )
       lengthOneDimension?                                           #dimensionDataType
     | typeName=(ENUM | SET)
-      '(' STRING_LITERAL (',' STRING_LITERAL)* ')' BINARY?
+      collectionOptions BINARY?
       (CHARACTER SET charsetName)? (COLLATE collationName)?         #collectionDataType
     | typeName=(
         GEOMETRYCOLLECTION | LINESTRING | MULTILINESTRING
         | MULTIPOINT | MULTIPOLYGON | POINT | POLYGON
       )                                                             #spatialDataType
     ;
 
+collectionOptions
+    : '(' collectionOption (',' collectionOption)* ')'
```

Comment:
Maybe it makes sense to simplify and replace `collectionOption` rule to `STRING_LITERAL`?

Reply:
It does not, I've pushed that change to this PR.